presto-blackhole/src/main/java/com/facebook/presto/plugin/blackhole/BlackHoleConnector.java

```diff
@@ -137,12 +147,27 @@ public ConnectorPageSinkProvider getPageSinkProvider()
                         value -> ImmutableList.copyOf(((List<String>) value).stream()
                                 .map(name -> name.toLowerCase(ENGLISH))
                                 .collect(toList())),
-                        List.class::cast));
+                        List.class::cast),
+                new PropertyMetadata<>(
+                        PAGE_PROCESSING_DELAY,
+                        "Sleep duration before processing the page",
+                        VARCHAR,
+                        Duration.class,
+                        Duration.valueOf("0s"),
+                        false,
+                        value -> Duration.valueOf((String) value),
+                        Duration::toString));
     }
 
     @Override
     public ConnectorNodePartitioningProvider getNodePartitioningProvider()
     {
         return partitioningProvider;
     }
+
+    @Override
+    public void shutdown()
+    {
+        shutdownAndAwaitTermination(executorService, 10, SECONDS);
```

Comment:
This can call `shutdownNow()` since `shutdown` is only called when the connector is no longer used